x-pack/plugins/files/common/types.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { SavedObjectAttributes } from '@kbn/core/server';
+
+export type FileStatus = 'AWAITING_UPLOAD' | 'UPLOAD_FAILED' | 'AVAILABLE';
```

Comment:
I'm thinking we might need one more status:

1. File metadata was created.
2. File contents upload has started (is in progress).
3. File uploaded successfully.
4. File uploaded with errors.

Something like:

```suggestion
export type FileStatus = 'CREATED' | 'UPLOADING' | 'READY' | 'ERROR';
```

Assuming we don't have any other errors other than an upload error.